inc/Modules/Rest/Governing_Controller.php

```diff
@@ -0,0 +1,319 @@
+				[
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => [ $this, 'set_github_token' ],
+					'permission_callback' => static fn (): bool => current_user_can( 'manage_options' ),
+					'args'                => [
+						'token' => [
+							'required'          => true,
+							'type'              => 'string',
+							'sanitize_callback' => 'sanitize_text_field',
+						],
+					],
+				],
+			],
+		);
+	}
+
+	/**
+	 * Get all public and private GitHub repositories from rtCamp and wpcomvip organizations.
+	 *
+	 * @return \WP_REST_Response|\WP_Error
+	 */
+	public function get_gh_repos(): \WP_REST_Response|\WP_Error {
+
+		// check into transient first.
+		$cached_repos = get_transient( self::TRANSIENT_GH_REPOS );
+		if ( false !== $cached_repos ) {
+			return rest_ensure_response(
+				[
+					'success' => true,
+					'repos'   => $cached_repos,
+					'count'   => count( $cached_repos ),
+				]
+			);
+		}
+
+		$github_token = Plugin_Settings::get_github_token();
+
+		if ( empty( $github_token ) ) {
+			return new \WP_Error( 'no_github_token', __( 'GitHub token not found.', 'oneupdate' ), [ 'status' => 404 ] );
+		}
+
+		// Loop to fetch all GitHub repos using pagination.
+		$all_repos = [];
+		$page      = 1;
+		$per_page  = 100;
+
+		do {
+			$fetch_url = "https://api.github.com/user/repos?affiliation=owner,organization,collaborator&per_page={$per_page}&page={$page}";
+
+			$response = wp_safe_remote_get(
+				$fetch_url,
+				[
+					'headers' => [
+						'Authorization' => 'Bearer ' . $github_token,
+						'Accept'        => 'application/vnd.github.v3+json',
+						'User-Agent'    => 'OneUpdate Plugin Loader',
+					],
+					'timeout' => 30, // phpcs:ignore WordPressVIPMinimum.Performance.RemoteRequestTimeout.timeout_timeout -- this is to avoid timeout issues.
+				]
+			);
+
+			if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
+				return new \WP_Error(
+					'github_api_error',
+					__( 'Failed to fetch GitHub repositories.', 'oneupdate' ),
+					[
+						'status'   => 500,
+						'error'    => is_wp_error( $response ) ? $response->get_error_message() : wp_remote_retrieve_response_code( $response ),
+						'response' => $response,
+					]
+				);
+			}
+
+			$repos = json_decode( wp_remote_retrieve_body( $response ), true );
+
+			if ( empty( $repos ) || ! is_array( $repos ) ) {
+				break;
+			}
+
+			$all_repos = array_merge( $all_repos, $repos );
+
+			++$page;
+
+			$repos_count = count( $repos );
+		} while ( $repos_count === $per_page );
+
+		// Filter for specific organizations.
+		$filtered_repos = [];
+		foreach ( $all_repos as $repo ) {
+			$filtered_repos[] = [
+				'slug' => $repo['full_name'],
+				'name' => $repo['name'],
+				'url'  => $repo['html_url'],
+			];
+		}
+
+		if ( empty( $filtered_repos ) ) {
+			return new \WP_Error( 'no_filtered_repos', __( 'No repositories found for rtCamp or wpcomvip.', 'oneupdate' ), [ 'status' => 404 ] );
```

Comment:
Is this specific `rtCamp` and `wpcomvip` callout on this error message intentional?